test/performance/performance/influx.go

```diff
@@ -0,0 +1,103 @@
+/*
+Copyright 2022 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package performance
+
+import (
+	"crypto/tls"
+	"fmt"
+	"log"
+	"os"
+	"time"
+
+	influxdb2 "github.com/influxdata/influxdb-client-go/v2"
+	"github.com/influxdata/influxdb-client-go/v2/api"
+)
+
+const (
+	influxToken = "INFLUX_TOKEN"
+	influxURL   = "INFLUX_URL"
+	prowTag     = "PROW_TAG"
+	org         = "Knativetest"
+	bucket      = "knative-serving"
+)
+
+// InfluxReporter wraps a influxdb client
+type InfluxReporter struct {
+	client   influxdb2.Client
+	writeAPI api.WriteAPI
```

Comment:
I guess the previous implementation was a bit inefficient as it was opening one connection per data point (AddInfluxPoint)?

Reply:
Exactly.